src/unix/os390-syscalls.c

```diff
@@ -308,6 +312,10 @@ int epoll_wait(uv__os390_epoll* lst, struct epoll_event* events,
     events[nevents++] = ev;
   }
 
+  if (msg_fd.revents != 0 && msg_fd.fd != -1)
+    if (i == lst->size)
```

Comment:
The loop guards on `i < maxevents` and `reventcount < pollret` so `i < lst->size` at least of some of the time?

Reply:
If `i < lst->size`, then the for loop hasn't reached the message queue FD and the corresponding event entry is not initialized, so there is no point in marking it as a message queue event.

Reply:
What if `maxevents < lst->size`? Won't that mean it'll skip marking the mq fd even though it's present?

Reply:
Yes, it will, but the queue should still have all unhandled messages the next time epoll_wait() is called. The original logic would do the same, since the for loop wouldn't reach the mq and the mq wouldn't be returned as part of the events array. I guess, it means if you have a lot of connections, all the file system events will get delayed...

Reply:
That seems like something that ought to be addressed? If not, can you explain why?

Reply:
I could address that but since it seemed to affect the old and new code the same way and wasn't related to the the original issue, I considered it out of scope :)

Reply:
Okay, fair enough.